ledger/participant-integration-api/src/test/lib/scala/platform/store/interning/InterningStringInterning.scala

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.interning
+
+import com.daml.lf.data.Ref
+import com.daml.lf.data.Ref.Party
+
+class InterningStringInterning extends StringInterning {
```

Comment:
Can we call it `MockStringInterningView` or some such. It took me a while to understand where to place this class in the string interning space.

Reply:
Why do we need this class? It looks like a simple implementation of the `StringInterning` trait. Can't we use `StringInterningView` instead? That one also implements `StringInterning` and lives in the same package.

RecoveringIndexerIntegrationSpec:
```
val stringInterning = new StringInterningView((_, _) => _ => Future.successful(Nil)) // not used
```

StorageBackendProvider:
```
val stringInterning = new InterningStringInterning
```

Reply:
I was going with MockStringInterning :ok_hand:

Reply:
@rautenrieth-da this is a step forward: this guy will be the workhorse for StorageBackend tests not caring about interning

Reply:
Why couldn't you use `StringInterningView` for those tests? It doesn't perform any IO, and is implemented in the same package.

Reply:
Unless I misunderstand the code, this is also not a mock - it's a full implementation of the `StringInterning` interface.

Reply:
> Why couldn't you use StringInterningView for those tests? It doesn't perform any IO, and is implemented in the same package.

We could, but I think we are better off if we don't because if we use the real thing, then in those tests we also will test the StringInterningView implementation, and also we have to wire to the test runner and to the tests that we populate correct entries there. (please note this MockStringInterning does not need maintenance: it is transparent from the test point of view)

> Unless I misunderstand the code, this is also not a mock - it's a full implementation of the StringInterning interface.

Correct it is not a mock in a strict sense, but I am ok with the naming because it implies test usage. It is a trivial, crude lightweight implementation of the real thing.
Have no strong opinions on the name, if you would have a better one, happy to take it :)